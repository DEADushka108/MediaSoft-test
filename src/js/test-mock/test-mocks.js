export * from './const/const';
export * from './ships/ships';
export * from './store/store';
