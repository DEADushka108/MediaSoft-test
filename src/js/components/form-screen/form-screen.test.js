import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import FormScreen from './form-screen';
import {store} from '../../test-mock/test-mocks';

it(`Should FormScreen component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <FormScreen/>
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
