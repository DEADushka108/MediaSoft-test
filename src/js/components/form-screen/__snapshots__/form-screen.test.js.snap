// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FormScreen component render correctly 1`] = `
Array [
  <header
    className="page-header"
  >
    <div
      className="page-header__wrapper"
    >
      <a
        className="page-header__logo"
        href="/"
        onClick={[Function]}
      >
        Starshipsshop
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link main-nav__link--active"
              href="/"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#home-page"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Shop
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/contact"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#social-group"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Contact
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/cart"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#shopping-cart"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Cart
              </span>
              <span
                className="main-nav__sub"
              >
                2
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <main
    className="page-main"
  >
    <h1
      className="visually-hidden"
    >
      Cart
    </h1>
    <div
      className="delivery"
    >
      <nav
        className="delivery__nav"
      >
        <ol
          className="delivery__list"
        >
          <li
            className="delivery__item delivery__item--active"
          >
            <p
              className="delivery__title"
            >
              1
               
              User
            </p>
          </li>
          <li
            className="delivery__item "
          >
            <p
              className="delivery__title"
            >
              2
               
              Card
            </p>
          </li>
          <li
            className="delivery__item "
          >
            <p
              className="delivery__title"
            >
              3
               
              Adress
            </p>
          </li>
        </ol>
      </nav>
      <section
        className="contact-us feedback"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <form
          className="feedback__form"
        >
          <fieldset
            className="feedback__item user-info"
          >
            <legend
              className="visually-hidden"
            >
              Introduce yourself an write us some word
            </legend>
            <div
              className="user-info__item"
            >
              <label
                className="user-info__label"
                htmlFor="user-name"
              >
                Full name
              </label>
              <input
                className="user-info__input"
                id="user-name"
                name="full-name"
                onChange={[Function]}
                placeholder="Your Name"
                required={true}
                type="text"
              />
            </div>
            <div
              className="user-info__item"
            >
              <label
                className="user-info__label"
                htmlFor="user-email"
              >
                Email
              </label>
              <input
                className="user-info__input"
                id="user-email"
                name="email"
                onChange={[Function]}
                placeholder="Your Email"
                required={true}
                type="email"
              />
            </div>
            <div
              className="user-info__item"
            >
              <label
                className="user-info__label"
                htmlFor="user-phone"
              >
                Phone
              </label>
              <input
                className="user-info__input"
                id="user-phone"
                name="phone"
                onChange={[Function]}
                placeholder="Your Phone"
                required={true}
                type="tel"
              />
            </div>
          </fieldset>
          <button
            className="feedback__button"
            onClick={[Function]}
            type="submit"
          >
            Next
          </button>
        </form>
      </section>
      <section
        className="payment"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <a
          className="feedback__button-back"
          onClick={[Function]}
        >
          <svg
            className="feedback__arrow-icon"
          >
            <use
              xlinkHref="#icon-right-arrow"
            />
          </svg>
          Back
        </a>
        <form
          className="payment__form"
        >
          <div
            className="payment__outro outro-number"
          >
            <p
              className="outro-number__text"
            >
              Card number
            </p>
            <p
              className="outro-number__text"
            >
              #### #### #### ####
            </p>
          </div>
          <div
            className="payment__outro outro-expire"
          >
            <p
              className="outro-expire__text"
            >
              Expires
            </p>
            <p
              className="outro-expire__text"
            >
              20/01
            </p>
          </div>
          <div
            className="payment__outro outro-name"
          >
            <p
              className="outro-name__text"
            >
              Card holder
            </p>
            <p
              className="outro-name__text"
            >
              Jonh Smith
            </p>
          </div>
          <img
            alt="Credit card"
            className="payment__img"
            src="./img/bg/credit-card-bg.png"
          />
          <fieldset
            className="payment__fieldset"
          >
            <legend
              className="visually-hidden"
            >
              Introduce yourself an write us some word
            </legend>
            <div
              className="payment__item"
            >
              <label
                className="payment__label"
                htmlFor="card-number"
              >
                Card number
              </label>
              <input
                className="payment__input"
                id="card-number"
                maxLength="16"
                minLength="13"
                name="card-number"
                onChange={[Function]}
                pattern="^[0-9]+$"
                placeholder="0000 00000 0000 0000"
                required={true}
                type="text"
              />
            </div>
            <div
              className="payment__item"
            >
              <label
                className="payment__label"
                htmlFor="card-name"
              >
                Card Name
              </label>
              <input
                className="payment__input"
                id="card-name"
                name="card-name"
                onChange={[Function]}
                placeholder="Jonh Smith"
                required={true}
                type="text"
              />
            </div>
            <div
              className="payment__item"
            >
              <div
                className="payment__expire"
              >
                <label
                  className="payment__label"
                  htmlFor="month"
                >
                  Expires
                </label>
                <input
                  className="payment__input payment__input-month"
                  id="month"
                  name="month"
                  onChange={[Function]}
                  required={true}
                  type="month"
                />
              </div>
              <div
                className="payment__cvv"
              >
                <label
                  className="payment__label"
                  htmlFor="card-cvv"
                >
                  Secret code
                </label>
                <input
                  className="payment__input payment__input-cvv"
                  id="card-cvv"
                  name="card-cvv"
                  placeholder="CVV"
                  required={true}
                  type="number"
                />
              </div>
            </div>
          </fieldset>
          <button
            className="payment__button"
            onClick={[Function]}
            type="submit"
          >
            Next
          </button>
        </form>
      </section>
      <section
        className="address"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <a
          className="feedback__button-back"
          onClick={[Function]}
        >
          <svg
            className="feedback__arrow-icon"
          >
            <use
              xlinkHref="#icon-right-arrow"
            />
          </svg>
          Back
        </a>
        <form
          className="address__form"
        >
          <fieldset
            className="address__fieldset"
          >
            <legend
              className="visually-hidden"
            >
              Introduce yourself an write us some word
            </legend>
            <div
              className="address__item"
            >
              <label
                className="address__label"
                htmlFor="adress"
              >
                Delivery address
              </label>
              <input
                className="address__input"
                id="address"
                name="address"
                onChange={[Function]}
                placeholder="Delivery Address"
                required={true}
                type="text"
              />
            </div>
            <div
              className="address__map"
            >
              <div />
            </div>
          </fieldset>
          <button
            className="feedback__button"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </form>
      </section>
    </div>
    ;
  </main>,
  <footer
    className="page-footer"
  >
    <div
      className="page-footer__wrapper"
    >
      <a
        className="page-footer__logo"
        href="/"
        onClick={[Function]}
      >
        Starshipsshop
      </a>
      <div
        className="page-footer__copyright"
      >
        <p
          className="page-footer__text"
        >
          © 2020 Stars ships shop Ltd.
        </p>
      </div>
    </div>
  </footer>,
]
`;
